### PR TITLE
V2.0 code review (do not merge)

### PR DESCRIPTION
**Pull request by Lorel**

### CODE REVIEW

File: frontend/index.template.html [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <title>{%=o.htmlWebpackPlugin.options.title || 'Redux React Router Async Example'%}</title>
```
Comment (robink) [inline frontend/index.template.html]: Should be updated with the real app title

File: frontend/index.template.html [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    <!--[if lt IE 9]>
+      <script src="http://cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7/html5shiv.js"></script>
+    <![endif]-->
+    <link rel="stylesheet" href="http://maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css">
+
+    {% for (var css in o.htmlWebpackPlugin.files.css) { %}
+    <link href="{%=o.htmlWebpackPlugin.files.css[css] %}" rel="stylesheet">
+    {% } %}
+
+    <script type="text/javascript" src="https://code.jquery.com/jquery-2.1.4.min.js"></script>
```
Comment (robink) [inline frontend/index.template.html]: Can we remove this now, as we are retrieving dependencies through npm

File: frontend/package.json [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  "name": "Slurm-web",
+  "version": "2.0.0",
+  "description": "",
+  "homepage": "https://github.com/edf-hpc/slurm-web",
+  "author": "Aurelien HAVET <aurelien@derniercri.io>",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/edf-hpc/slurm-web.git"
+  },
+  "license": "GNU General Public License version 3",
```
Comment (robink) [inline frontend/package.json]: should be checked against the project license

File: frontend/src/components/Application.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
+import React, { PropTypes } from 'react'
+// import classnames from 'classnames'
```
Comment (robink) [inline frontend/src/components/Application.js]: remove if not needed

File: frontend/src/components/Footer.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+export default class Footer extends React.Component {
+
+  render () {
+    return (
+      <div className="footer">
+        <div className="pure-g">
+          <div className="pure-u-1 u-sm-1-2">
+            This site is built with {'<3'} using
+            {' '}
+            <a href="http://purecss.io/" target="_blank">PureCSS</a>
```
Comment (robink) [inline frontend/src/components/Footer.js]: should be removed or updated

File: frontend/src/components/MenuListItem.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  renderLink () {
+    if (this.props.isExternal)
+      return (
+        <a href={this.props.link} target="_blank" className="pure-menu-link">
+          {this.props.text}
+        </a>
+      )
+    else
+      return (
+        <Link to={this.props.link} className="pure-menu-link">
```
Comment (robink) [inline frontend/src/components/MenuListItem.js]: do we need this class?

File: frontend/src/components/elements/plots/Plots.js [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+    //     .attr('height', options.height)
+    //     .attr('id', target + '-svg')
+    //
+    //   svg.append('g')
+    //
+    //   donut3d.draw(target + '-svg', datas,
+    //     options.width / 2, options.height / 2,
+    //     options.width * 0.4, options.height * 0.3, 20, 0)
+    // } else
+    //   donut3d.transition(target + '-svg', datas, 100, 60, 20, 0)
```
Comment (robink) [inline frontend/src/components/elements/plots/Plots.js]: comments should be removed

File: frontend/src/components/elements/racks/draw.js [hunk 8]
```diff
@@ -0,0 +1,564 @@
...
+    drawRect(ctx, legendX - 2, legendY, 9, 9, colors.colorFullyAllocated)
+    ctx.fillStyle = 'black'
+    ctx.fillText('fully allocated', legendX + 10, legendY + 10)
+
+    legendY += 10
+    drawRect(ctx, legendX - 2, legendY, 9, 9, colors.colorPartAllocated)
+    ctx.fillStyle = 'black'
+    ctx.fillText('partly allocated', legendX + 10, legendY + 10)
+  }
+}
```
Comment (robink) [inline frontend/src/components/elements/racks/draw.js]: feel like this file is a little too big

File: frontend/src/components/elements/table/Row.js [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+          </td>
+        )
+      })
+
+    return (
+      <tr
+        id={
+          header ?
+          config.ids.thead.tr(datas) :
+          config.ids.tbody.tr(datas)
```
Comment (robink) [inline frontend/src/components/elements/table/Row.js]: improve readibility by doing this (`xx ? yy : zz`)  before the return statement

File: frontend/src/components/elements/table/Table.js [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+export default class Table extends React.Component {
+
+  static propTypes = {
+    datas: PropTypes.array.isRequired,
+    config: PropTypes.object.isRequired,
+    onRowClick: PropTypes.func
+  }
+
+  render () {
+
```
Comment (robink) [inline frontend/src/components/elements/table/Table.js]: remove empty line

File: frontend/src/components/elements/table/Table.js [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+          datas={data}
+          config={config}
+          onClick={onRowClick}
+        />)
+    })
+
+    let columns = config.columns.map( (column) => {
+      return column.name
+    })
+
```
Comment (robink) [inline frontend/src/components/elements/table/Table.js]: remove one of the empty line

File: frontend/src/components/pages/Jobs.js [hunk 12]
```diff
@@ -0,0 +1,144 @@
...
+})
+@Radium
+export default class Jobs extends React.Component {
+
+  // statics
+  // constructor
+  // lifecycle methods (willmount...)
+  // handlers
+  // others
+  // render@
```
Comment (robink) [inline frontend/src/components/pages/Jobs.js]: remove theses

File: frontend/webpack.config.js [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+    filename: 'app.js',
+    path: path.join(__dirname, 'dist'),
+    publicPath: '/assets/'
+  },
+  plugins: [
+    new webpack.HotModuleReplacementPlugin(),
+    new webpack.NoErrorsPlugin(),
+    new ExtractTextPlugin('app.css', { allChunks: true }),
+    new HtmlWebpackPlugin({
+      title: 'Redux React Router Async Example',
```
Comment (robink) [inline frontend/webpack.config.js]: update the title

File: frontend/src/components/elements/racks/draw.js (hunk 8 above)
Comment (Lorel) [inline frontend/src/components/elements/racks/draw.js]: yeah, right
deep refacto needed

**Pull request closed without merging**